src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,320 @@ def inner(tree, path, root_path):
+                if query in api_objects[most_similar_api_object_index]:
+                    print(api_objects[most_similar_api_object_index])
+            else:
+                most_similar_api_object_indices = cosine_similarities.argsort()
+                for most_similar_api_object_index in reversed(
+                    most_similar_api_object_indices[0][-3:]
+                ):
+                    if query in api_objects[most_similar_api_object_index]:
+                        print(api_objects[most_similar_api_object_index])
+
+
+def _get_wildcard_matches_for_word_from_names(word: str, names: list):
+    """Get wildcard matches for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    wildcard_matches: list
+        Matched API object names.
+    """
+    wildcard_matches = [name for name in names if fnmatch.fnmatch(name, word)]
+    valid_wildcard_matches = [
+        wildcard_match for wildcard_match in wildcard_matches if wildcard_match in names
+    ]
+    return valid_wildcard_matches
+
+
+def _search_wildcard(search_string: str):
+    """Perform wildcard search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+
+    Returns
+    -------
+        List of search string matches.
+    """
+    names = _get_api_object_names()
+    queries = _get_wildcard_matches_for_word_from_names(search_string, names)
+    if queries:
+        _print_search_results(queries, wildcard=True)
+
+
+def _get_exact_match_for_word_from_names(
+    word: str,
+    names: list,
+):
+    """Get exact match for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    exact_matche: list
+        List of exact match.
+    """
+    return [name for name in names if word == name]
+
+
+def _get_close_matches_for_word_from_names(
+    word: str,
+    names: list,
+):
+    """Get close matches for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    valid_close_matches: list
+        List of valid close matches.
+    """
+    close_matches = closest_allowed_names(word, names)
+    valid_close_matches = [
+        close_match for close_match in close_matches if close_match in names
+    ]
+    return valid_close_matches
+
+
+def _search_whole_word(search_string: str, match_case: bool):
+    """Perform exact search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
```

Comment:
```suggestion
        Word to search for. Semantic search is the default.
    match_case: bool
        Whether to match case. If ``True``, the search is case-insensitive.
```
True to a ``match_case`` parameter to me implies that the search is case-sensitive. Is what's here a mistake?

Reply:
No. It will match both upper and lower case words.

Reply:
@hpohekar Maybe it wasn't clear. The default answer to the Boolean question (Whether to match case) is ``True``--which in my mind says yes, it's true, match case. For a search non-sensitive to matching case, it seems like the default should be ``False``. In other words: If ``False``, the search is case-insensitive.

Reply:
@PipKat 

If `match_case=True` means we are trying to match capitalize word. i.e. case-sensitive. 

for e.g font -> Font or Font -> Font

If `match_case=False` means we are trying to match whole word. i.e, case-insensitive.

for e.g RemovePartitionLinesTolerance -> RemovePartitionLinesTolerance

![image](https://github.com/ansys/pyfluent/assets/106588300/12d7c763-9e51-4d8e-a00d-d309cbda5fb4)